### ISSUE TITLE
Horizon: account stream sends too many events and closes connection for no reason

### ISSUE THREAD

**Issue by andrenarchy** [stellar/go#735]
Streaming an account via `/accounts/{id}` is unreliable at the moment:

* The connection is constantly dropped by Horizon for no reason
* It keeps sending events even though nothing happened (1 message every 2 seconds)

Here's an example where you can see the issue:
```
curl -v -N -H "Accept:text/event-stream" https://horizon-testnet.stellar.org/accounts/GC772KV23BEEC4WU6IH7AYNJKCGEWSB7P3F7HWYILCCF6KN3KTQJJQPR\?cursor\=now\&limit\=200
```

**Comment by bartekn** [comment]
This behaviour haven't changed in a while, horizon-testnet.stellar.org is currently running 0.15.0 release candidate but if you test it in the public network it will work in the same way.

Single object streams (like `/account/{id}`) are updated every second. In the next version we're planning to add "update interval" parameter and other stream improvements.

**Comment by bartekn** [comment]
> What about the dropped connections? Shouldn't we keep this issue open until this has been addressed?

Horizon drops the connection when the limit of events has been reached, it `/accounts/{id}`, it's equal `10`: https://github.com/stellar/go/blob/master/services/horizon/internal/actions_account.go#L48

> Also, I'm not sure why it needs to send the same data over and over again – an update interval is not really helping, it should just stream the events as they happen, right?

I agree that streaming should be improved. We're going to release Horizon 0.15.0 today that fixes some of the streaming issues (ex. adds heartbeat) but let's aim to fix other streaming issues in 0.16.0.

**Comment by tomquisel** [comment]
@andrenarchy if we only send events when data has changed as in #828, will that fully address this issue, or is there still an incorrect dropped connections issue?

**Comment by andrenarchy** [comment]
@tomquisel Issue #828 only seems to be related to orderbooks and not to account data, is this right?

**Comment by bartekn** [comment]
@andrenarchy it has been changed to support `/accounts/{id}` too.

**Comment by bartekn** [comment]
Fixed in #828.
